File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sum = 0
        
        if len(nums)<3:
            sum = nums[0]+nums[1]
            if sum == target:
                return [0,1]
        else:
            for i in range(len(nums)):
                diff = target -nums[i]
                if diff in nums and nums.index(diff)!=i:
                    return [i,nums.index(diff)]
            
        
                
    
                
            
            
            
        

File: two-sum/two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Hashtable<Integer,Integer> ht = new Hashtable<>();
        int diff = 0,index=0;;
        int[] res = new int[2];
        for(int i=0;i<nums.length;i++){
            diff = target-nums[i];
            if(ht.containsKey(diff)){
                res[index] =i; 
                index++;
                res[index] = ht.get(diff);
            }
            else{
                ht.put(nums[i],i);
            }
        }
        return res;
        
    }
}